test/performance/service_creation_test.go

```diff
@@ -0,0 +1,255 @@
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nsName,
+			},
+		}
+		_, err := clients.KubeClient.CoreV1().Namespaces().Create(context.Background(), ns, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("Failed to create namespace %q: %#v", nsName, err)
+		}
+	}
+	defer cleanupNamespaces(t, clients, namespaces)
+
+	t.Log("Generating services by kperf.")
+	// kperf service generate -n 10 -b 30 -c 5 -i 15 --namespace default  --svc-prefix ktest --wait --timeout 30s --max-scale 3 --min-scale 0
+	generateCmd := exec.Command("kperf", "service", "generate", "-n", strconv.Itoa(servicesCount),
+		"-i", "10", "-b", "10",
+		"--min-scale", "1", "--max-scale", "2",
+		"--namespace-prefix", nsNamePrefix,
+		"--namespace-range", servicesRange,
+		"--svc-prefix", KperfFlags.ServiceNamePrefix,
+		"--wait", "true",
+		"--timeout", servicesTimeout.String())
+	var out bytes.Buffer
+	generateCmd.Stdout = &out
+	err := generateCmd.Run()
+	if err != nil {
+		t.Fatalf("Failed to execute kperf service generate command %#v", err)
+	}
+	fmt.Println(out.String())
+
+	t.Log("Waiting for Services.")
+	index := 0
+	for _, ns := range namespaces {
+		servingClientInNamespace, err := servingClientForNamespace(ns)
+		if err != nil {
+			t.Fatalf("Failed to get serving client %#v", err)
+		}
+		if err := v1test.WaitForServiceState(servingClientInNamespace, fmt.Sprintf("%s-%d", KperfFlags.ServiceNamePrefix, index), v1test.IsServiceReady, "ServiceIsReady"); err != nil {
+			t.Fatalf("Failed waiting for serving to get Ready %#v", err)
+		}
+		index++
+	}
+
+	t.Log("Measuring results by kperf.")
+	output := KperfFlags.KperfOutput
+	if output == "" {
+		output, err = os.MkdirTemp("", "kperfoutput")
+		if err != nil {
+			t.Fatalf("Failed creating temp folder %#v", err)
+		}
+		defer os.RemoveAll(output)
+	}
+
+	// kperf service measure --namespace kref-test-1 --svc-prefix ksvc --range 0,9  --verbose --output /tmp
+	measureCmd := exec.Command("kperf", "service", "measure",
+		"--verbose",
+		"--namespace-prefix", nsNamePrefix,
+		"--namespace-range", servicesRange,
+		"--svc-prefix", KperfFlags.ServiceNamePrefix,
+		"--range", servicesRange,
+		"--output", output)
+
+	var stdoutBuf, stderrBuf bytes.Buffer
+	measureCmd.Stdout = io.MultiWriter(os.Stdout, &stdoutBuf)
+	measureCmd.Stderr = io.MultiWriter(os.Stderr, &stderrBuf)
+
+	err = measureCmd.Run()
+	if err != nil {
+		t.Fatalf("Failed to execute kperf service measure command %#v", err)
+	}
+	outStr, errStr := string(stdoutBuf.Bytes()), string(stderrBuf.Bytes())
+	fmt.Printf("\nout:\n%s\nerr:\n%s\n", outStr, errStr)
+
+	measurment := MeasureResult{}
+	err = readResultFromJsonFile(output, "", &measurment)
+	if err != nil {
+		t.Fatalf("Failed to parse kperf service measurment result %#v", err)
+	}
+
+	if measurment.Result.OverallAverage > KperfFlags.ServiceAverage {
+		t.Fatal("kperf Services creation took too long")
+	}
+
+	if KperfFlags.UploadLatestResults {
```

Comment:
since this is used in both tests, would this make sense as a helper function?

Reply:
its calling a helper func, but after checking if the flag is set